google-cloud-vision/lib/google/cloud/vision/v1.rb

```diff
@@ -49,7 +51,7 @@ module Vision
       # ```rb
       # require "google/cloud/vision"
       #
-      # image_annotator_client = Google::Cloud::Vision.new(version: :v1)
+      # image_annotator_client = Google::Cloud::Vision::ImageAnnotator.new(version: :v1)
```

Comment:
This approach is not going to scale. There is no guarantee different API versions will have the same client types. I would prefer to see an optional `type` argument here that specified the client to return. 

In truth, I would prefer to see an improved GAPIC client that could wrap multiple stubs and present all API methods.

Reply:
It's a tricky issue. I actually prefer to see the separate modules because these are conceptually separate services (under the umbrella of "Vision"). So I'd argue Ruby developers would expect to see different "classes" to instantiate.

Reply:
I can see the argument for having them at the version level, but not the level above it. e.g. `Google::Cloud::V1::ImageAnnotator.new`, but not `Google::Cloud::ImageAnnotator.new(version: :v1)`

Reply:
I think the difference between specifying the type and specifying the version is that the former is absolutely required, whereas the latter is optional and indeed probably left at default by most users. Thus, it makes some sense for there to be syntax indicating "what service" you are instantiating (i.e. make it look like a class instantiation) while keeping version as an optional named argument that most people don't pay attention to.

As for the multiple-stub-wrapping client, that might be difficult to document.

Reply:
This may or may not be helpful, but for the sake of discussion, the Ads team took an approach that is similar to what @blowmage is describing with a single entry point (i.e. vision) to lookup clients (i.e. image_annotator) via a symbol: 

https://github.com/googleads/google-ads-ruby/blob/981a9e78eeaa87ebb2b670c1975f3e9a6a46ed72/lib/google/ads/google_ads/google_ads_client.rb#L82 

In their case, version isn't an argument but there's a good reason for it that doesn't apply to here. Might be worth exploring both options for the new baselines @quartzmo and @landrito?

Reply:
I think this part of the interface definitely worth exploring for improvements!

Reply:
Either way, this is a breaking change (either a different module or a required parameter). Since we are also in the middle of moving vision to GAPIC-only, it's probably worth making a decision here on what the interface ought to look like.